sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs

```diff
@@ -192,10 +192,18 @@ internal static T ToTableEntity<T>(this IDictionary<string, object> entity, Prop
                     }
                     else
                     {
+                        if (propertyValue is null)
+                        {
+                            continue;
+                        }
                         if (property.PropertyType.IsEnum)
                         {
                             typeActions[typeof(Enum)](property, propertyValue, result);
                         }
+                        else if (property.PropertyType.Name.Equals("Nullable`1") && property.PropertyType.GenericTypeArguments.Length == 1 && property.PropertyType.GenericTypeArguments[0].IsEnum)
```

Comment:
```suggestion
                        else if (Nullable.GetUnderlyingType(property.PropertyType)?.IsEnum == true)
```

Reply:
I knew I must have been missing something like this - thanks!